From 7431f80dd3f16330b75198a3699c4512f72300d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Wed, 1 Oct 2025 12:57:44 +0200
Subject: [PATCH 371/553] Pass though size correctly in [NSImage
 imageFromQIcon] overload

Pick-to: 6.8 6.5
Change-Id: I7fe8a778976ff388d447500e53fa4633e4883b10
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 0de42247d34247a7c8ad1475444f4245f79eb37e)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/gui/painting/qcoregraphics.mm | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/gui/painting/qcoregraphics.mm b/src/gui/painting/qcoregraphics.mm
index 0917f2484db..f605cc741d5 100644
--- a/src/gui/painting/qcoregraphics.mm
+++ b/src/gui/painting/qcoregraphics.mm
@@ -124,10 +124,9 @@ QT_END_NAMESPACE
 
 + (instancetype)imageFromQIcon:(const QIcon &)icon withSize:(int)size
 {
-    return [NSImage imageFromQIcon:icon withSize:0 withMode:QIcon::Normal withState:QIcon::Off];
+    return [NSImage imageFromQIcon:icon withSize:size withMode:QIcon::Normal withState:QIcon::Off];
 }
 
-
 + (instancetype)imageFromQIcon:(const QIcon &)icon withSize:(int)size withMode:(QIcon::Mode)mode
                                                                      withState:(QIcon::State)state
 
-- 
2.51.2

